lib/record.rb

```diff
@@ -0,0 +1,20 @@
+class Record
+  attr_reader :amount, :date, :type
+
+  def initialize(amount, date)
+    @amount = check_validity(amount)
+    @date = date
+    @type = set_type
+  end
+
+  private
+
+  def check_validity(amount)
+    raise '0 is not a valid amount' if amount.zero?
+    amount
+  end
+
+  def set_type
```

Comment:
A bit of a nitpick, but it doesn't quite feel like `set_type` actually sets the type.  Rather, it gets it or evaluates it or something like that.